linera-client/src/chain_listener.rs

```diff
@@ -85,57 +85,68 @@ pub trait ClientContext: 'static {
 
 /// A `ChainListener` is a process that listens to notifications from validators and reacts
 /// appropriately.
-pub struct ChainListener {
-    config: ChainListenerConfig,
-    listening: Arc<Mutex<HashSet<ChainId>>>,
+pub struct ChainListener<C: ClientContext> {
+    context: Arc<Mutex<C>>,
+    storage: C::Storage,
+    config: Arc<ChainListenerConfig>,
+    listening: Arc<std::sync::Mutex<HashSet<ChainId>>>,
 }
 
-impl ChainListener {
+impl<C: ClientContext> Clone for ChainListener<C> {
```

Comment:
Isn't this `#[derive(Clone)]`?

Reply:
Only if `C: Clone`, which we don't actually need.